src/services/identity/UsersService.ts

```diff
@@ -79,7 +79,11 @@ class UsersService {
     return !!siteMember
   }
 
-  async findSitesByUserId(isomerId: string) {
+  async findSitesByUserId(
+    isomerId: string
+  ): Promise<User & { site_members: Site[] }> {
```

Comment:
can i check if you've verified this? the collaborators PR types this as `SiteMembers` (and i found that to be true at runtime) 

separately, since i think we'll be using this alot (eg: collaborators PR), do you think this merits an extra type? 

nit: once retrieved, we should transform this into `siteMembers`

Reply:
Hmm I don't think it should be the SiteMembers table, as the return value should have the `repo` property, which we use to obtain the name of the repository (refer to `getSitesForEmailUser()` under SitesService.ts).

I updated this in f7045aa as it should match the type for the `sites` property in User.ts.

As for the collaborators PR I think it's slightly different. This one is from the User.ts perspective, whereas the other one is from the Sites.ts property, so the types are inherently different.

Reply:
good point on `User vs Site`. also, i realised (by dint of experimentation and subsequent investigation) that if we have the correct annotations on the respective model, the base model (`User` here) should contain the foreign key **object** (ie, `site_members`).

Hence, this casting might be indicative of a erroneous assocation or lack thereof... **but** if you verified that this is working, we can ignore this (for now) and fix it later (perhaps on friday?)

Reply:
I am guessing the error came because of how we used the `as` operator to declare it as `site_members`, but the actual key is `sites` here:

https://github.com/isomerpages/isomercms-backend/blob/398d70f8dc46d97edd672f4d0df04400b580e8f4/src/database/models/User.ts#L64-L70

It's a hypothesis that the key should be renamed to `site_members` so we would not have this kind of props issues, similar to how it is done for Site.ts. I think I can look into this further beyond this PR, to see if simply renaming the prop is possible.

Reply:
for future reference, this is correct and the link to the docs is [here](https://github.com/sequelize/sequelize-typescript#type-safe-through-table-instance-access)